### ISSUE TITLE
Command history with cursor up in internal shell

### ISSUE THREAD

**Issue by ulope** [inducer/pudb#348]
When inside the internal shell the cursor-up key just selects the lines in history with no (apparent) useful function. 

It would be nice if it could instead access the history (as does ctrl-p/n).

**Comment by inducer** [comment]
While I understand the request, I fear that that would be inconsistent with how navigation in Urwid works. Under your suggested change, Ctrl-X would likely be the only way to navigate away from the (internal) command line.

**Comment by asmeurer** [comment]
I find this annoying too. You can navigate history in the internal shell with Ctrl-p / Ctrl-n. Going through the history is useful, especially for long outputs like tracebacks. But I would prefer if the arrow keys defaulted to history.

**Comment by inducer** [comment]
Hmm, OK. I cold be convinced, if one of you were to produce a PR.

**Comment by subwindow** [comment]
Agree, this is super annoying. It's really messing up my muscle memory.

**Comment by ned2** [comment]
Perhaps this could be a configurable preference?

up/down going back/forward through shell history is fundamentally hardwired into my expectations of how REPLs work. I'm loving PUDB but this is breaking my brain.

What would be involved in putting together a PR for this?

**Comment by asmeurer** [comment]
IMO it should just be changed. Should be a simple PR.

**Comment by inducer** [comment]
Given the overwhelming sentiment on this, I'm on board.

**Comment by ned2** [comment]
I've created submitted a PR for this at #392. I figured leaving the existing ctrl p/n bindings made sense?
